src/main/java/org/opentripplanner/openstreetmap/model/OSMWithTags.java

```diff
@@ -281,16 +281,28 @@ public boolean isPedestrianExplicitlyAllowed() {
     }
 
     /**
-     * Returns true if through traffic is not allowed.
-     * 
+     * Returns true if through traffic for motor vehicles is not allowed.
+     *
+     * @return
+     */
+    public boolean isMotorVehicleThroughTrafficExplicitlyDisallowed() {
+        String motorVehicle = getTag("motor_vehicle");
+        return isGeneralNoThroughTraffic() || "destination".equals(motorVehicle);
+    }
+
+    public boolean isBicycleNoThroughTrafficExplicitlyDisallowed() {
+        String bicycle = getTag("bicycle");
+        return isGeneralNoThroughTraffic() || "destination".equals(bicycle);
+    }
+    /**
+     * Returns true if through traffic for bicycle is not allowed.
+     *
      * @return
      */
-    public boolean isThroughTrafficExplicitlyDisallowed() {
+    public boolean isGeneralNoThroughTraffic() {
         String access = getTag("access");
-        boolean noThruTraffic = "destination".equals(access) || "private".equals(access)
-                || "customers".equals(access) || "delivery".equals(access)
-                || "forestry".equals(access) || "agricultural".equals(access);
-        return noThruTraffic;
+        return "destination".equals(access) || "private".equals(access)
+                || "customers".equals(access) || "delivery".equals(access);
```

Comment:
It does seem strange, but do you know why the "forestry" and "agricultural" tags where checked here?

Reply:
We also thought that this is strange - if not a bug. In Germany at least forestry and agricultural roads are not no-thru by default. They maybe closed to non-farming traffic but that's not what this piece of code did.

For that reason I removed it.

If no-thru really is specific to each country then we need move that into the way property set.

Reply:
It might be that it is there to avoid taking shortcuts through forestry and agricultural roads, even if it is technically allowed. We could talk about this on the next dev meeting, to get other people's input.

Reply:
Good idea.

Reply:
Handling of the tags was moved to `WayPropertySetSource` so that is may be modified as-needed, along with a mention of the changes to the mailing list.